### PR TITLE
Add ANSI C12.22 protocol dissector

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

This protocol is used to communicate with electric meters. PCAP example is taken from the Wireshark wiki.

### CODE REVIEW

File: src/lib/protocols/c1222.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+static void ndpi_search_c1222(struct ndpi_detection_module_struct *ndpi_struct,
+                              struct ndpi_flow_struct *flow)
+{
+  struct ndpi_packet_struct const * const packet = &ndpi_struct->packet;
+
+  NDPI_LOG_DBG(ndpi_struct, "search ANSI C12.22\n");
+
+  if ((packet->payload_packet_len < 70) || (packet->payload[0] != 0x60) ||
+      ((u_int16_t)(packet->payload_packet_len-2) != packet->payload[1]))
```
Comment (IvanNardi) [inline src/lib/protocols/c1222.c]: (u_int8_t)

**Review by IvanNardi — CHANGES_REQUESTED** [review]
Just a minor issue.
This is  a trace with udp traffic [ansic12.12_udp.zip](https://github.com/ntop/nDPI/files/14285216/ansic12.12_udp.zip)
Could you double check that is is properly classified (it seems so to me) and merge part of it with your tcp trace, please?

**Comment by 0xA50C1A1** [comment]
> Just a minor issue. This is a trace with udp traffic [ansic12.12_udp.zip](https://github.com/ntop/nDPI/files/14285216/ansic12.12_udp.zip) Could you double check that is is properly classified (it seems so to me) and merge part of it with your tcp trace, please?

You found the UDP trace? Thank you so much.

Yeah, I'll fix all the issues in a moment.

**Review by IvanNardi — APPROVED** [review]
Thanks (also for removing the skype leftovers...)

**Pull request merged**